GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3077,154 +3072,215 @@ private void LoadSubmodulesIntoDropDownMenu()
-                    smi.Image = Resources.IconFolderSubmodule;
+                    var smi = new SubmoduleInfo { Text = name, Path = path };
+                    result.OurSubmodules.Add(smi);
+                    GetSubmoduleStatusAsync(smi, cancelToken);
                 }
 
+                if (threadModule.SuperprojectModule != null)
                 {
-                    var name = "Superproject: ";
-                    GitModule parentModule = Module.SuperprojectModule;
-                    string localpath = "";
-                    if (Module.SuperprojectModule.WorkingDir != supersuperproject.WorkingDir)
+                    GitModule supersuperproject = threadModule.FindTopProjectModule();
+                    if (threadModule.SuperprojectModule.WorkingDir != supersuperproject.WorkingDir)
                     {
-                        parentModule = supersuperproject;
-                        localpath = Module.SuperprojectModule.WorkingDir.Substring(supersuperproject.WorkingDir.Length);
-                        localpath = PathUtil.GetDirectoryName(localpath.ToPosixPath());
-                        name = name + localpath;
-                    }
-                    else
-                        name = name + Path.GetFileName(Path.GetDirectoryName(supersuperproject.WorkingDir));
-                    string path = Module.SuperprojectModule.WorkingDir;
-                    if (Settings.DashboardShowCurrentBranch && !GitModule.IsBareRepository(path))
-                        name = name + " " + GetModuleBranch(path);
-
-                    var smi = AddSubmoduleToMenu(name, Module.SuperprojectModule);
-                    GetSubmoduleStatusImageAsync(smi, parentModule, localpath);
-                }
+                        var name = "Top project: " + Path.GetFileName(Path.GetDirectoryName(supersuperproject.WorkingDir));
+                        string path = supersuperproject.WorkingDir;
+                        if (Settings.DashboardShowCurrentBranch && !GitModule.IsBareRepository(path))
+                            name = name + " " + GetModuleBranch(path);
 
-                var submodules = supersuperproject.GetSubmodulesLocalPathes().OrderBy(submoduleName => submoduleName);
-                if (submodules.Any())
-                {
-                    string localpath = Module.WorkingDir.Substring(supersuperproject.WorkingDir.Length);
-                    localpath = PathUtil.GetDirectoryName(localpath.ToPosixPath());
+                        result.TopProject = new SubmoduleInfo { Text = name, Path = supersuperproject.WorkingDir };
+                        GetSubmoduleStatusAsync(result.TopProject, cancelToken);
+                    }
 
-                    foreach (var submodule in submodules)
                     {
-                        var name = submodule;
-                        string path = supersuperproject.GetSubmoduleFullPath(submodule);
+                        var name = "Superproject: ";
```

Comment:
This was a problem in the original code.  I'll add a separate commit to fix this.